pyne/mesh.py

```diff
@@ -155,7 +156,7 @@ def __getitem__(self, key):
             return getattr(mats[int(key)], name)
         elif isinstance(key, slice):
             return np.array([getattr(mats[i], name) for i in range(*key.indices(size))])
-        elif isinstance(key, np.ndarray) and key.dtype == np.bool:
+        elif isinstance(key, np.ndarray) and key.dtype in _BOOLEAN_TYPES:
```

Comment:
I believe that setting `key.dtype == bool` may be more efficient in this case. This comparison checks if the `data type (dtype)` of key is exactly `bool`. On the other hand, using `key.dtype in (bool, np.bool_)` checks if the dtype is either `bool` or `np.bool_`. This approach involves iterating over the elements of the tuple to find a match, which could potentially be slower.

Reply:
You are right that my proposed change would be technically slower, involving 2 equality checks instead of one, but note that
- we're talking 10s of nano-seconds here (hopefully this doesn't matter)
- because `bool` is the *first* element in `_BOOLEAN_TYPES`, the second check is only performed in case `key.dtype != bool`
- if performance was important here (again, I'm hoping it's not), `key.dtype is bool` would be the way to go (identity check is always cheaper than equality check, although here it's 20ns VS 24ns on my machine)

Reply:
Considering the potential impact on simulation time when the code is called multiple times (considering large data), it becomes more important to optimize the comparison. In that case, using `key.dtype == bool` would be more efficient compared to `key.dtype in (bool, np.bool_)` due to the reduced number of operations involved. However, it's still advisable to measure the actual performance gain before making a decision, as the difference might still be minimal.

Reply:
Does the proposed approach ensure some kind of backward compatibility?  Are there configurations that may use one vs the other?  Or is `bool` always available so that any invocation will always work if we restrict to `bool`?  Put another way, what's the benefit to this proposed approach?

Reply:
My approach is supposed to be 100% backward compatible. Additionally, it survives `np.bool_` (current implementation doesn't)